packages/dynamodb-data-mapper/src/DataMapper.ts

```diff
@@ -306,6 +307,100 @@ export class DataMapper {
         }
     }
 
+    private delay(duration: number) {
+        return new Promise(resolve => setTimeout(resolve, duration));
+    }
+
+    /**
+     * Perform a UpdateTable operation using the schema accessible via the
+     * {DynamoDbSchema} property and the table name accessible via the
+     * {DynamoDbTable} property on the prototype of the constructor supplied.
+     *
+     * The promise returned by this method will not resolve until the table is
+     * active and ready for use.
+     *
+     * @param valueConstructor  The constructor used for values in the table.
+     * @param options           Options to configure the CreateTable operation
+     */
+    async createGlobalSecondaryIndex(
+        valueConstructor: ZeroArgumentsConstructor<any>,
+        {
+            readCapacityUnits,
+            streamViewType = 'NONE',
+            writeCapacityUnits,
+            indexOptions = {},
+        }: CreateTableOptions,
+        indexName: string
+    ) {
+        const schema = getSchema(valueConstructor.prototype);
+        const { attributes, indexKeys, tableKeys } = keysFromSchema(schema);
+        const TableName = this.getTableName(valueConstructor.prototype);
+
+        const globalSecondaryIndexes = indexDefinitions(indexKeys, indexOptions, schema).GlobalSecondaryIndexes;
+        const indexSearch = globalSecondaryIndexes.filter(function(index) {
+            return index.IndexName === indexName;
+        });
+        const indexDefinition: CreateGlobalSecondaryIndexAction = indexSearch[0];
+
+        var {
```

Comment:
Nitpicky: use `let` instead of `var` for consistency.

Reply:
Thanks for catching this. With the other changes, it can be const again. I realized after digging through the aws-sdk code that it was already poll-waiting for the 'ACTIVE' state. The table returns to an 'ACTIVE' state while the new GSI is in a backfilling state, so this doesn't guarantee that .query against the new GSI will succeed right away.

Reply:
Would it be worth polling `DescribeTable` until all GSIs were no longer backfilling?